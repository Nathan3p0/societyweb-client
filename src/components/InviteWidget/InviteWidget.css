.inviteWidget__container {
    background: #fff;
    margin: 1rem 1rem;
    box-shadow: 1px 3px 10px #33333363;
}

.inviteWidget__container-header {
    background-color: #26495E;
    color: white;
    padding: .3rem;
    padding-left: 1rem;
    font-size: 1.5rem;
}

.inviteWidget__container-content {
    padding: 1rem;
    font-size: 1.1rem;
}

.inviteWidget__form li {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    margin-top: .5rem
}

.inviteWidget__form > li > label {
    flex: 1 0 100px;
    font-size: 1.5rem;
}

.inviteWidget__form > li > label + * {
    flex: 1 0 220px;
}

.inviteWidget__form li button {
    margin-left: auto;
    padding: 8px 16px;
    border: none;
    background: #333333;
    color: #f2f2f2;
    text-transform: uppercase;
    letter-spacing: .09em;
    border-radius: 2px;
}

.inviteWidget__form li button:hover {
    cursor: pointer;
    transform: scale(1.1);
    box-shadow: 1px 1px 5px #333333;
}

.inviteWidget__form li input {
    padding: .5rem;
    border: 1px solid #33333341;
}