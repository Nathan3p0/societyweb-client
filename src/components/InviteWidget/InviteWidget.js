import React, { Component } from 'react';
import AdminApiService from '../../services/admin-api-service';
import './InviteWidget.css';

class InviteWidget extends Component {
    state = {
        error: null,
        success: false
    }

    validatePhoneNumber = (phone) => {
        const phoneRegEx = /^\(?(\d{3})\)?[- ]?(\d{3})[- ]?(\d{4})$/;

        return phoneRegEx.test(phone);
    }

    handleInviteSubmit = (e) => {
        e.preventDefault();

        const { phone } = e.target;
        const { value } = phone;

        if (!this.validatePhoneNumber(value)) {
            this.setState({
                error: 'Please enter a valid phone number'
            })
        } else {

            AdminApiService.inviteNewMemberSms(value)
                .then(res => {
                    phone.value = '';
                    this.setState({
                        success: true
                    })
                })
                .catch(res => {
                    this.setState({
                        error: res.error
                    })
                })
        }
    }

    render() {
        const { error, success } = this.state
        return (
            <section className="inviteWidget__container">
                <h3 className="inviteWidget__container-header">Invite</h3>
                <div className="inviteWidget__container-content">
                    <p>Simply enter your new member's phone number and we will send them an invite.</p>
                    <form onSubmit={this.handleInviteSubmit}>
                        <ul className="inviteWidget__form">
                            {error && <p className="error">{error}</p>}
                            {success && <p className="success">Your invite was sent successfully!</p>}
                            <li>
                                <label htmlFor="invite">Phone Number:</label>
                                <input type="tel" name="phone" id="invite" placeholder="123-555-0505" required />
                            </li>
                            <li>
                                <button type="submit">Send Invite</button>
                            </li>
                        </ul>
                    </form>
                </div>
            </section>
        );
    }
}

export default InviteWidget;