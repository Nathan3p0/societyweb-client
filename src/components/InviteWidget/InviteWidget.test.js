import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import InviteWidget from './InviteWidget'

describe('Invite Widget Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<InviteWidget />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<InviteWidget />).toJSON();
        expect(tree).toMatchSnapshot()
    })
})