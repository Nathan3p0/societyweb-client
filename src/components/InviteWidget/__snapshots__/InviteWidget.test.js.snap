// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invite Widget Component Rendering Tests renders to UI as expected 1`] = `
<section
  className="inviteWidget__container"
>
  <h3
    className="inviteWidget__container-header"
  >
    Invite
  </h3>
  <div
    className="inviteWidget__container-content"
  >
    <p>
      Simply enter your new member's phone number and we will send them an invite.
    </p>
    <form
      onSubmit={[Function]}
    >
      <ul
        className="inviteWidget__form"
      >
        <li>
          <label
            htmlFor="invite"
          >
            Phone Number:
          </label>
          <input
            id="invite"
            name="phone"
            placeholder="123-555-0505"
            required={true}
            type="tel"
          />
        </li>
        <li>
          <button
            type="submit"
          >
            Send Invite
          </button>
        </li>
      </ul>
    </form>
  </div>
</section>
`;
